[dvelop] Merge forward from fluorine to develop

rallytime opened the pull request:
Conflicts:
  - salt/modules/virt.py
  - tests/filename_map.yml
  - tests/unit/states/test_virt.py

terminalmage committed:
Add --from-filenames and --filename-map options to parser

This allows for the passed filename(s) to decide which tests are run

terminalmage committed:
Add an initial filename_map

terminalmage committed:
Don't run unit or integration tests unnecessarily

If --from-filenames (and optionally --filename-map) yield no matches,
this will keep from running the entire unit/integration suite.

terminalmage committed:
Enhancements to --from-filenames, --filename-map options

This commit adds the following features:

- Changes to test modules are identified, and those tests are added to
  the list of tests to be run
- expr_match is used to allow the filename map to use either globbing or
  regular expressions
- Corresponding state tests are added when doing filename matches,
  meaning that for example the git state tests will be run when the git
  execution module is in the --from-filenames list

terminalmage committed:
Fill out the filename map a bit

terminalmage committed:
Add more unit test mapping

terminalmage committed:
Move helper outside of for loop

terminalmage committed:
Don't import old module name

We'll handle the old name when backporting

cbosdo committed:
Rename states/test_libvirt.py into states/test_virt.py

Modules test files are named after the module itself, the state virt
module needs to comply with that rule too.

terminalmage committed:
Rename test pgjsonb returner test file to match naming convention

terminalmage committed:
Move jinja templating tests alongside salt/utils/jinja.py's tests

terminalmage committed:
Rename test_map_conf test file to respect naming convention

terminalmage committed:
Consolidate and move loader tests to test/unit/test_loader.py

terminalmage committed:
Consolitdate and move salt-ssh unit tests to proper location

terminalmage committed:
Consolidate saltclass pillar unit tests into single module

terminalmage committed:
Rename/consolidate salt.utils unit tests to conform to naming convention

terminalmage committed:
Rename custom_grains.py so it's not identified as a test module

terminalmage committed:
Rename some files to avoid them being identified as test modules

terminalmage committed:
Rename providers to clouds to respect naming convention

Also, move rackspace openstack-based tests to test_openstack.py

terminalmage committed:
Add function to get all of the test modules

terminalmage committed:
Use the new helper to get the test mods

terminalmage committed:
Add additional paths to the filename map

terminalmage committed:
Add back test accidentally deleted when cherry-picking

terminalmage committed:
Add back changes lost in cherry-pick

terminalmage committed:
Rename files to match naming convention

terminalmage committed:
Add integration.netapi.rest_tornado.test_app to filename map

terminalmage committed:
Add test to check test module names and make sure they match the naming convention

terminalmage committed:
Make sure we always run the module names test

terminalmage committed:
Lint

terminalmage committed:
Moar Lint

terminalmage committed:
Evan Moar Lint

terminalmage committed:
Fix conflicting dict key in filename map

terminalmage committed:
Add documentation explaining naming convention for test files

terminalmage committed:
Set a default value for the filename map path

Commit:
Add some doc tests to the filemap that should run on every PR

These files test aspects of the documentation that are not covered by the
pylint or doc build runs. Therefore, these should run on all incoming PRs.

gtmanfred committed:
enable testing only filemaps

terminalmage committed:
runtests.py: Accept modified file list from a text file

dwoz committed:
Make sure from-filenames intersect with names-file

When both names-file an from-names are defined we want to make sure the
tests that from-filenames returns intersect with the test in the names
file.

terminalmage committed:
runtests.py: Fix IndexError in --from-filenames

ethanculler committed:
Added handling for minion return exceptions containing the word "return"

Commit:
Remove test files for beacons that don't exist in 2018.3

ethanculler committed:
Merge branch '2017.7' into fix-minion-return-exception-with-return

Ch3LL committed:
Update search engine IDs for releases

Ch3LL committed:
change order of search engine ids

Commit:
Merge pull request #50233 from Ch3LL/bp-50232

[2017.7] Backport #50232

gtmanfred committed:
bump salt version to 2017.7.8

Commit:
Add init file back in for integration/cloud/clouds pkg

This was accdientally removed during some cloud test file renaming, which
prevents the test from running due to an ImportError

Commit:
Remove test_error from scheduler tests

These tests are not present on 2018.3

garethgreenaway committed:
Porting #50216 to fluorine.

Commit:
Remove "test_custom" from grains section of filename_map

This test is for functionality only available in fluorine and newer.

Removing this for 2018.3.

Commit:
Remove pgjsonb returner unit tests

The tests that are written in this file are for functions not present
in the 2018.3 branch.

This commit will need to be reverted in the merge-forward to `fluorine`.

gtmanfred committed:
Merge pull request #50251 from gtmanfred/2017.7

bump salt version to 2017.7.8 for test bootstrapping

Commit:
Adding securitygroupname as the config instead of securitygroup

gtmanfred committed:
Merge pull request #50258 from dubb-b/fix_cloud_ec2

Adding securitygroupname as the config instead of securitygroup

gtmanfred committed:
Merge pull request #50184 from ethanculler/fix-minion-return-exception-with-return

Added handling for minion return exceptions containing the word "return"

m03 committed:
Fix missing aptly config paths

terrible-broom committed:
apt: speedup installing packages with '>=' or '<=' and reduced memory consumption

Commit:
Merge pull request #50267 from m03/fix-aptly-config-paths-20183

Fix missing aptly config paths

Commit:
Merge branch '2017.7' into '2018.3'

No conflicts.

Commit:
Add init file back in for unit/client pkg

This file was accidentally removed during some test renaming. The absence
of this file prevents the test from running due to an ImportError.

Commit:
Use get_config_dir() instead of self.config_dir

This attribute does not exist in the 2018.3 branch. Use the function
for now, and then change this back on the merge-forward.

Commit:
Remove saltclass pillar unit test for list expansion

This test only applies to functionality present in Fluorine or newer.

This commit will need to be reverted again in the fluorine branch.

Commit:
Revert "Rename test_map_conf test file to respect naming convention"

This reverts commit 8326a09921a3f5211d75f1c24075fc6c721e2039.

Commit:
Lint fix: add extra blank line in core.py

gtmanfred committed:
fix InstallRequirement.from_line for pip 18.1

Fixes #49967

Commit:
Adding subnetID so it knows the right VPC to build the node in.

Commit:
Merge branch '2017.7' of https://github.com/saltstack/salt into add_subnet

Commit:
Revert "Move jinja templating tests alongside salt/utils/jinja.py's tests"

This reverts commit 6d9c2eb6b0884c6625da781e234b7dc896f23fae.

Commit:
Fixing lint error

Commit:
Lint: Add extra blank line

Commit:
Merge pull request #50281 from dubb-b/add_subnet

Add subnet

Commit:
Merge branch '2017.7' into '2018.3'

cbosdo committed:
virt: expose source_file disk property to user

User may want to update or create a VM with a disk image created or
provided by a third party tool. In order to do this, let the user set
a source_file property to the disk.

Thus a {'name': 'myimage', 'source_file': '/path/to/image'} disk will
just use the provided disk image.

Commit:
Merge pull request #50182 from terminalmage/issue50050

Backport runtests.py modifications from develop to 2018.3

twangboy committed:
Mock is_windows = False

Commit:
Merge branch '2018.3' into 'fluorine'

Conflicts:
  - tests/unit/states/test_virt.py
  - tests/unit/test_config.py

cbosdo committed:
virt: changing disk bus should update it in the VM

So far two disks are considered equal if they have the same source file.
When a user only wants to change the disk bus, then the disk comparison
algorithm will consider it unchanged.

Now disks are compared on source and target bus. The other disk
variables can be left out side the target device is computed.

Commit:
Revert "Remove test files for beacons that don't exist in 2018.3"

This reverts commit dfa05b978eaad93dcbfd620b5896c7d59a7b59e3.

Commit:
Revert "Remove test_error from scheduler tests"

This reverts commit 68146983cbeb6d842c1bb1b13a180dcab4dafec9.

Commit:
Revert "Remove "test_custom" from grains section of filename_map"

This reverts commit 68ab3eef55fa0a812d0aa74645facc9955bf08c3.

Commit:
Revert "Remove pgjsonb returner unit tests"

This reverts commit ab4a670ff22878d5115f408baf0304a0ba3ec994.

Commit:
Revert "Remove saltclass pillar unit test for list expansion"

This reverts commit 85a9395c2a2d2ab2d7e1e159cb741b4d096d4015.

gtmanfred committed:
Merge pull request #50280 from gtmanfred/2017.7

fix InstallRequirement.from_line for pip 18.1

Commit:
minor correction to boto_vpc.route_table_present example, subnet_names was on wrong indent level

Commit:
Merge pull request #50287 from whytewolf/boto_vpc

minor correction to boto_vpc.route_table_present example

Commit:
Merge pull request #50282 from twangboy/fix_test_diskusage

Fix `unit.beacons.test_diskusage` on Windows

Commit:
Merge pull request #50276 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

gtmanfred committed:
if line in output of localectl is empty, ignore it

gtmanfred committed:
add extra space to localemod test

gtmanfred committed:
fix test

Commit:
Remove beacon unit tests that end with *_beacon.py

These have all been renamed for the filename_map functionality.

Commit:
Remove test_pgjsonb_return unit test file

This file has been renamed for the filename_map functionality.

Commit:
Remove the test_saltclass_list_expasion unit test file

The tests contained in this file have been moved to the test_saltclass
unit test file. This is necessary for filename_map to work correctly.

Commit:
Remove the test_custom_grains integration file

This file has been renamed for the filename_map functionality.

Commit:
Merge pull request #50283 from rallytime/merge-fluorine

[fluorine] Merge forward from 2018.3 to fluorine

terrible-broom committed:
fix review notes

gtmanfred committed:
update jenkins version for 2018.3

Commit:
Removing an un needed change and maybe fixing the lint error

gtmanfred committed:
Merge pull request #50289 from gtmanfred/2018.3

ignore empty line in localectl status output

twangboy committed:
Add missing return to secretbox_decrypt()

gtmanfred committed:
Merge pull request #50293 from saltstack/jenkins

update jenkins version for 2018.3

Commit:
Merge pull request #50285 from cbosdo/fluorine-backport

virt disk fixes for fluorine

gtmanfred committed:
mark ansiblegate test as flaky

Commit:
Lint: add extra blank line

Commit:
Merge pull request #50270 from terrible-broom/feature/apt-cache-show-instead-of-dump

apt: speedup installing packages with '>=' or '<=' and reduced memory…

Commit:
Merge pull request #50299 from gtmanfred/ansible

mark ansiblegate test as flaky

Commit:
Merge pull request #50298 from twangboy/fix_nacl_util

Add missing return to secretbox_decrypt()

Commit:
Fix matcher grain file

This was inadvertently changed in th filename_map backport.

This commit restores the matcher grain file to match all branches previous
to the inadvertent change.

Commit:
Merge pull request #50295 from dubb-b/remove-cloud-change

Removing an un needed change and maybe fixing the lint error

Commit:
Merge pull request #50308 from rallytime/test-fixes

Fix matcher grain file

Commit:
Merge branch '2017.7' into '2018.3'

Conflicts:
  - salt/states/pip_state.py
  - tests/integration/cloud/clouds/test_ec2.py

Commit:
Merge pull request #50309 from rallytime/merge-2018.3

[2018.3] Merge forward from 2017.7 to 2018.3

Commit:
Merge branch '2018.3' into 'fluorine'

Conflicts:
  - salt/modules/aptpkg.py
  - tests/integration/cloud/clouds/test_ec2.py

Commit:
Merge pull request #50256 from garethgreenaway/porting_50216_to_fluorine

[fluorine] Porting #50216 to fluorine

Commit:
Merge pull request #50312 from rallytime/merge-fluorine

[fluorine] Merge forward from 2018.3 to fluorine

Commit:
Merge branch 'fluorine' into 'develop'

Conflicts:
  - salt/modules/virt.py
  - tests/filename_map.yml
  - tests/unit/states/test_virt.py

gtmanfred commented on .kitchen.yml:
```diff
@@ -1,6 +1,6 @@
 ---
 <% vagrant = system('gem list -i kitchen-vagrant 2>/dev/null >/dev/null') %>
-<% version = '2017.7.6' %>
+<% version = '2018.3.3' %>
```
We need to make sure the salt-jenkins is merged forward to master for this.

rallytime replied:
@gtmanfred Ok can I get a review here?

https://github.com/saltstack/salt-jenkins/pull/1187

Then I can merge-forward to `master` in salt-jenkins.

Commit:
Fix virt unit tests

About 7 were failing after a bad merge forward.

This commit restores the setup functions and mocked class needed for
the tests to run.

Commit:
Remove debugging comment

Commit:
Lint: Remove extra blank line

Commit:
Remove test class that is already present

The `SaltclassPillarTestCaseListExpansion` was already defined in develop.
The merge forward added the class again, even though the code was the same.

This removes the double class definition.
